src/pages/de.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * A backwards compatibility page for redirecting legacy links
+ * to their proper current locations.
+ *
+ * @author psarando
+ */
+import React from "react";
+
+import { useRouter } from "next/router";
+
+import { getAppLaunchPath, getQuickLaunchPath } from "components/apps/utils";
+import { getFolderPage } from "components/data/utils";
+import systemId from "components/models/systemId";
+import BasicTable from "components/utils/BasicTable";
+
+export default function DeLegacyRedirector() {
+    const router = useRouter();
+    const query = router.query;
+
+    let redirectPath;
+    switch (query?.type) {
+        case "apps":
+            redirectPath = getAppLaunchPath(
+                query["system-id"],
+                query["app-id"]
+            );
+            break;
+
+        case "data":
+            redirectPath = getFolderPage(query.folder);
+            break;
+
+        case "quick-launch":
+            redirectPath = getQuickLaunchPath(
+                query["system-id"] || systemId.de,
+                query["app-id"],
+                query["quick-launch-id"]
+            );
+            break;
+
+        default:
```

Comment:
Nice ✨ 

I did a quick check in the old DE and looks like we also had the case for an app category query string in the path.  Not sure how often that was used or if it's worth preserving at this point, but figured I'd mention it: https://github.com/cyverse-de/ui/blob/master/de-lib/src/main/java/org/iplantc/de/desktop/client/presenter/DesktopPresenterImpl.java#L812-L817

Reply:
Good catch. I forgot to double check the old code.

It looks like it's attempting to open the apps listing to the appropriate category, and it may have attempted to select the correct app with the given app ID as well, though it's not doing that anymore.

The new code can display the apps listing for a specific category with a query param like `selectedCategory={"system_id":"de","id":"..."}`, except the category filter drop-down will display a blank label.

If that's acceptable for these legacy links, I can add that change to this PR as well.

Reply:
How is this different from https://de.cyverse.org/de/?type=apps&app-id=67d15627-22c5-42bd-8daf-9af5deecceab&system-id=de? This link also filters by `system-id` and `id` ? 🤔

Reply:
The link https://de.cyverse.org/de/?type=apps&app-id=67d15627-22c5-42bd-8daf-9af5deecceab&system-id=de
will redirect users to https://de.cyverse.org/apps/de/67d15627-22c5-42bd-8daf-9af5deecceab/launch (since the old DE is opening the launch dialog for those links now), but for a link like https://de.cyverse.org/de/?type=apps&system-id=de&app-category=8835fcff-53d8-472a-8fc5-9ffc03ebe925, I was going to add a redirect to the apps listing page, such as: `https://de.cyverse.org/apps?selectedCategory={"system_id":"de","id":"8835fcff-53d8-472a-8fc5-9ffc03ebe925"}`.

That will open the apps listing for the given category, and it will also ignore any `app-id` query param (which is what the old DE is also doing now).

Reply:
Ahh now I see it. Somehow, I got confused with `id` for selected category with app id. 👍🏽 .